### ISSUE TITLE
Incorrect kernel is calculated for homomorphism

### ISSUE THREAD

**Issue by sTertooy** [gap-packages/polycyclic#40]
The following bug appears in the current master, but not in the latest release (v2.15.1). I'm afraid I don't see how to reduce this to a simpler example.
```
gap> H := ExamplesOfSomePcpGroups( 15 );;
gap> G := H/LowerCentralSeriesOfGroup( H )[8];;
gap> g := (G.2^-2*G.3^3*G.5^2*G.8^-6*G.9^45*G.10^-24);;
gap> srcs := [ G.2*G.5^-1, G.3, G.6, G.7, G.8, G.9, G.10 ];;
gap> M := Subgroup( G, srcs );;
gap> N := LowerCentralSeriesOfGroup( G )[7];;
gap> idG := Identity( G );;
gap> imgs := [ G.9^-3*G.10^-3, G.9^-2, G.9^4, G.10^-2, idG, idG, idG ];;
gap> diff := GroupHomomorphismByImages( M, N, srcs, imgs);;
gap> Image( diff, g );
id
gap> g in Kernel( diff );
false
```
Clearly, `g` should be in the kernel. The first isomorphism theorem also doesn't hold for the calculated kernel and image:
```
gap> M / Kernel( diff );
Pcp-group with orders [ 0, 2, 0, 0 ]
gap> Image( diff );
Pcp-group with orders [ 0, 0 ]
```

**Comment by fingolfin** [comment]
Thank you for your report! I observe that the kernel is not correct. On `master`, we get:
```
gap> Pcp(Kernel(diff));
Pcp [ g3^2*g6, g8, g9, g10 ] with orders [ 0, 0, 0, 0 ]
```
While in 2.15.1 we have
```
gap> Pcp(Kernel(diff));
Pcp [ g2^2*g3*g5^-2*g6^2*g7^-4, g3^2*g6, g8, g9, g10 ] with orders [ 0, 0, 0, 0, 0 ]
```

Using `git bisect`, I determined that this regression was introduced by commit 901f1a5ae52754d4e6399c4609bce98be8c3db18 which @beick made to fix another issue in `AddToIgs`.

**Comment by sTertooy** [comment]
This seems to be the same issue as #38 , then. Perhaps it's best to close this duplicate issue, then?

**Comment by fingolfin** [comment]
I plan to look into both. Your additional datapoint might be helpful in debugging it, so let's leave it open for now, it doesn't hurt :-).

**Comment by fingolfin** [comment]
Fixed via #43
